### ISSUE TITLE
FanControl will not take control of UniHub speeds

### ISSUE THREAD

**Issue by Astondb9driver** [EightB1ts/FanControl.LianLi#8]
FanControl will not take control of fan speeds. It did not pair the speeds automatically on setup and can read the speeds properly when I change in L-Connect 3, but I can not change speeds in FanControl. I have unblocked the plugin if that info helps. I do not know how to to get the device manger info because I just get this:

![image](https://github.com/EightB1ts/FanControl.LianLi/assets/139933567/8846927b-acbc-42d7-bd57-752865d02a8b)

In viewing 'ALL EVENTS':
![image](https://github.com/EightB1ts/FanControl.LianLi/assets/139933567/c0fa2c5c-3a3f-495d-b653-fca08b102d1b)

(Do not know if that will help)

Here is what my FanControl looks like.
![image](https://github.com/EightB1ts/FanControl.LianLi/assets/139933567/a7e34d8d-43e0-4236-b365-115701a2beb9)

**Comment by Tinka007** [comment]
Helllo,

I also have the same problem as described above. Fan Control shows the correct RPM but control doesn't work. A few days ago it worked fine. After installingen Fan Control again and setting up the plugin it seems to be not working properly.

![Screenshot 2023-07-22 120710](https://github.com/EightB1ts/FanControl.LianLi/assets/140244459/7b9ae61d-c059-42de-9db9-1ebe1d0ff282)

**Comment by EightB1ts** [comment]
Could you both share your log file located in in the same directory as the FanControl executable? Thanks

**Comment by Tinka007** [comment]
Here are my logs. I've looked myself but i don't see anything from Lian Li plugin itself. Their are multiple ones from the NZXT plugin, you can ignore these.

[log Fan Control.txt](https://github.com/EightB1ts/FanControl.LianLi/files/12136489/log.Fan.Control.txt)

**Comment by EightB1ts** [comment]
Something is probably going on with the NZXT plug-in. I assume @Astondb9driver also has that installed because I see an NZXT device in his device manager screenshot. Could you try removing the NZXT plug-in and see if the LianLi one starts working?

**Comment by Tinka007** [comment]
I already removed the NZXT plugin. I also reinstalled fan control and only used the Lian Li plugin but without any improvement.
Feel free to let me know if you need additional info. Would be nice to get it working as intended.

I also noticed the Controller option in the Fan Control software, this is currently enabled.

![Screenshot 2023-07-22 151419](https://github.com/EightB1ts/FanControl.LianLi/assets/140244459/1001ca51-6510-4564-86c3-8ec751e2bac3)

**Comment by EightB1ts** [comment]
Could you delete the log file and rerun the reinstalled FanControl NZXT with plugin removed? Then share the new log file. Thanks.

**Comment by Tinka007** [comment]
Thanks for the reply, i did the following.

- Deleted Fan Control compelely, restarted my computer.
- Unpacked Fan Control software and ran the detection setup at first startup
- Closed Fan Control and put the LianLi DLL (Non ARGB) file in the plugin folder.  Unblocked it via properties
- Started Fan Control, no error messages. No log files was created, i assume because no error messages.
Note: i didn't enable the Controller in the settings when doing the detection setup. Not sure if this needs to be enabled.

As you can see, Fan Control detects the fans and shows the correct RPM. I verified this in Lian Li Connect-3 software.
I put a flat curve of 100% on the Channel 1 but RPM stays at configured curve from the Lian Li software.

![Screenshot 2023-07-22 153259](https://github.com/EightB1ts/FanControl.LianLi/assets/140244459/af638c7c-5a64-40be-ade0-7972342756b6)

**Comment by EightB1ts** [comment]
Do you keep L-connect running in the background? Looking at the fan mode in L-connect, it’s configured to keep sending fan speed commands to the controller. That seems like a separate issue.

@Astondb9driver It looks like you have a Kraken too. Are you getting this same error? This will cause those symptoms you experienced. Which is an issue with that plugin.

```22/07/2023 12:10:14: Unhandled exception in FanControl v164.0.0.0
22/07/2023 12:10:14: System.NullReferenceException: Object reference not set to an instance of an object.
   at FanControl.NzxtKraken.NzxtKrakenX3.Update()
   at FanControl.NzxtKraken.NzxtKrakenPlugin.Update()
   at FanControl.Domain.BackendProviders.Plugin.PluginBackendProvider.Update()
   at System.Collections.Generic.List`1.ForEach(Action1 action)
   at FanControl.Domain.ComputerAccessLayer.Update()
```

**Comment by Tinka007** [comment]
L-Connect is running in the background. I just closed L-Connect and able to change the RPM in Fan Control, did not knew this...
The NZXT kraken plugin gives issue's aswell, Send the issue to the maker of the plugin. Many thanks for taking your time to help out. Appreciate it!

**Comment by Astondb9driver** [comment]
Hi, sorry. I've been on a trip the past week. I do not have any plugins installed beside your Lian Li plugin. I also was not running L-Connect in the background when having these issues so I'm not sure I'm having the same issue as Tinka007 (?).
